### PR TITLE
Docks and nested shuttles & Removes some major sleepers (100% lag free)

### PR DESCRIPTION
**Pull request by dwasint**
# About The Pull Request
also reverts updates our mc to tg's as its needed for this
https://github.com/BeeStation/BeeStation-Hornet/pull/7152
https://github.com/BeeStation/BeeStation-Hornet/pull/7659

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog

:cl:
add: Nested shuttles. shuttles can now land on top of each other and be towed around.
add: Dock turfs. Shuttles can always land on these, regardless of whether the dock is in space, on the station, or on another shuttle. Tiles are craftable with 1 plasteel per 4 tiles.
add: Dry docks. Works like docks, but you can build on these as if you were in space and designate what you build onto a shuttle. Tiles are craftable with 2 plasteel per 4 tiles.
tweak: Arrival shuttles and Emergency shuttles cannot be towed by other shuttles.
refactor: Lots of changes to shuttle code, especially relating to docking.
balance: Machinery now blocks custom docking to prevent theft of important equipment.
/:cl:

### CODE REVIEW

File: code/controllers/master.dm [hunk 1]
```diff
@@ -97,27 +86,27 @@ GLOBAL_REAL(Master, /datum/controller/master) = new
...
 			Recover()
 			qdel(Master)
 			Master = src
 		else
 			//Code used for first master on game boot or if existing master got deleted
 			Master = src
-			var/list/subsystem_types = subtypesof(/datum/controller/subsystem)
-			sortTim(subsystem_types, /proc/cmp_subsystem_init)
+			var/list/subsytem_types = subtypesof(/datum/controller/subsystem)
+			sortTim(subsytem_types, /proc/cmp_subsystem_init)
```
Comment (nednaZ) [inline code/controllers/master.dm:98]: Spelling error

**Comment by PowerfulBacon** [comment]
The MC revert isn't needed as the changes to the MC were for debugging purposes and can be removed (There is a debugging define which turns the freeze monitoring on). That MC seemed to cause us noticeably more lag when merged though, so we reverted it and everyone in discord told me it got better. I have no hard evidence, but almost everyone said the same thing.

**Comment by nednaZ** [comment]
Converted to draft as it is not running through tests

**Comment by dwasint** [comment]
this has a bunch of extra work to be workable gonna reopen in the future

**Pull request closed without merging**